### PR TITLE
 Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by ProgramKingAbel**
# THE n+1 FIX :hammer_and_pick: 

## BEFORE EAGERLOAD ON POST#INDEX ACTION 

before_eager_load_index

## AFTER EAGERLOAD ON POST#INDEX ACTION 

after_eager_load_index

## BEFORE EAGERLOAD ON POST#SHOW ACTION

before_eager_load_show

## AFTER EAGERLOAD ON POST#SHOW ACTION 

after_eager_load_show

# Integration tests with rspec and Capybara

## TEST RESULTS
test results

### User index page:

You can see the username of all other users, the profile picture for each user,  the number of posts each user has written.
When you click on a user, you are redirected to that user's show page.

### user show page:

You can see the user's profile picture, the user's username, the number of posts the user has written, the user's bio, the user's first 3 posts, a button that lets you view all of a user's posts.

When you click a user's post, it redirects to that post's show page.
When you click to see all posts, it redirects to the user's post's index page.

### User post index page:

You can see the user's profile picture, the user's username, the number of posts the user has written.
A post's title, see some of the post's body,  the first comments on a post, how many comments a post has, how many likes a post has, a section for pagination if there are more posts than fit on the view.
When post is clicked, it redirects me to that post's show page.

### Post show page:

You can see the post's title, who wrote the post, how many comments it has, how many likes it has, the post body, the username of each commentor, the comment each commentor left.

### CODE REVIEW

File: spec/views/users/index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      expect(page).to have_css("img[src*='https://www.rd.com/wp-content/uploads/2020/11/RD-mickey-mouse-GettyImages-1137134597-JVcrop.jpg']")
+    end
+  end
+
+  it 'redirects to the user\'s show page' do
+    visit user_path(user1)
+    expect(page).to have_current_path(user_path(user1))
+    expect(page).to have_content(user1.name)
+    expect(page).to have_content(user1.bio)
+  end
```
Comment (codepantha) [inline spec/views/users/index_spec.rb]: - [ ] This test should assert that you redirect to the user's show page when you click on a user, however you are already visiting the user's show page when you did `visit user_path(user 1)`. Please rewrite this test to capture the user interaction from 
1. clicking a user,
2. asserting that the current path is the user's show page, 
3. asserting that the page has content pertaining to that user.

File: spec/views/posts/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      expect(page).to have_content(comments[0].author.name)
+      expect(page).to have_content(comments[1].author.name)
+      expect(page).to have_content(comments[2].author.name)
+    end
+    it 'shows posts comments' do
+      expect(page).to have_content(comments[0].text)
+      expect(page).to have_content(comments[1].text)
+      expect(page).to have_content(comments[2].text)
+    end
+  end
```
Comment (codepantha) [inline spec/views/posts/show_spec.rb]: - [ ] Please write tests that asserts you can see a post's body 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Wrote integration specs for views 📌 
✔️ All tests are green ✔️ 
✔️ No linter errors 🟢 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @ProgramKingAbel && @lorenzoworx  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Made changes as requested by the previous reviewer :+1: 
- All written tests passing :heavy_check_mark: 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**